src/examples/hotgym/HelloSPTP.cpp

```diff
@@ -115,33 +115,32 @@ Real64 BenchmarkHotgym::run(UInt EPOCHS, bool useSPlocal, bool useSPglobal, bool
 
     //Encode
     tEnc.start();
-    enc.encodeIntoArray(r, input.data());
+    enc.encodeIntoArray(r, input.getSparse().data());
+    input.setSparseInplace(); //update SDR
```

Comment:
made these public to allow above.

Reply:
Please dont make the inplace setters public.  Instead just call the regular setters, they will detect if they need to copy or not.  If you hand a setter method the correct datatype the it will swap vector data instead of copying.  The following line will update the SDR data inplace:
`setDense( sdr.getDense() )`

Reply:
should we provide SDR::update(int type) ? to make this more clear

Reply:
I think th